components/accountPreferences/PasswordPreferences.tsx

```diff
@@ -0,0 +1,108 @@
+import { usePasswordChangeMutation } from "@/saleor/api";
+import React from "react";
+import { useForm } from "react-hook-form";
+
+export const PasswordPreferences: React.VFC<any> = ({}) => {
+  const [passwordChangeMutation] = usePasswordChangeMutation({});
+  const [successMessage, setSuccessMessage] = React.useState<String | null>(
+    null
+  );
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setError,
+    clearErrors,
+  } = useForm();
+
+  const onPasswordPreferenceSubmit = handleSubmit(async (formData) => {
+    if (formData.newPassword !== formData.newPasswordRepeat) {
+      setError("error", { message: "Passwords have to match." });
+    } else {
+      const result = await passwordChangeMutation({
+        variables: {
+          newPassword: formData.newPassword,
+          oldPassword: formData.oldPassword,
+        },
+      });
+      const errors = result.data?.passwordChange?.errors || [];
+      if (errors.length > 0) {
+        errors.forEach((e) => setError("error", { message: e.message || "" }));
+        setTimeout(() => {
+          clearErrors();
+        }, 3000);
+        return;
```

Comment:
```suggestion
errors.forEach((e) =>
        setError(e.field as keyof PasswordChangeFormData, {
          message: e.message || "",
        })
      );
      return;
```

Reply:
When we assign errors to the fields, error messages will be automatically hidden when the field is edited. Thats the default behavior of react-hook-form